### PR TITLE
feat: budget-vs-actual alert notifications

### PR DESCRIPTION
**Pull request by digaomatias**
## Summary

Proactively notifies users when a budget category **approaches** (default 80%) or **exceeds** its limit. The notable part: almost all the infrastructure already existed and was unused — the `BudgetThreshold`/`BudgetExceeded` notification types, the budget-progress calculation, the per-user `BudgetAlertPercentage` preference, rate limiting, and quiet hours were all already in place. This PR connects them.

## How it works

- **Trigger** (`INotificationTriggerService.CheckBudgetThresholdsAsync`): loads the user's active budgets, reuses the same `CalculateBudgetProgressAsync` the budget UI uses, and fires one notification per category that crosses the threshold.
  - *Exceeded takes precedence* over approaching (only the most severe fires).
  - *Unbudgeted categories* (effective budget ≤ 0) are skipped so users aren't spammed.
  - *Per-user threshold* honoured, defaulting to 80%.
- **Idempotency**: the `groupKey` is scoped to the budget **period** (`budget:{id}:cat:{id}:threshold:{approaching|exceeded}:{periodStart}`), so a user gets exactly one approaching + one exceeded alert per category per period — not one per day the job runs.
- **Job** (`BudgetAlertJobService`): daily Hangfire job at **1:30 AM** (after the 1:00 AM expired-budget rollover, so it evaluates the current period). Backed by a new `IBudgetRepository.GetUserIdsWithActiveBudgetsAsync`.
- **Settings**: a "Budget Alerts" card on `/settings` lets users set the threshold (1–100%). It saves the full preference set so the threshold change doesn't reset other notification settings (the backend replaces all fields).
- **Bell**: `BudgetThreshold`/`BudgetExceeded` templates (en + pt-BR). The bell already deep-links via `data.href` → `/budgets/{id}`, so no component change was needed.

No schema migration — `BudgetAlertPercentage` already existed.

## Test plan
- [x] `dotnet build` + `npm run build` clean
- [x] Backend starts clean with the new Hangfire job registered ("No pending migrations")
- [x] **12 new unit tests** — trigger (approaching / exceeded / precedence / below-threshold / zero-budget / per-user threshold / period-scoped groupKey / no-active-budgets) and job (one-call-per-user / no-users / per-user-error-isolation)
- [x] Full unit suite green (1097 passed)
- [x] Settings "Budget Alerts" card verified live — loads current value, saves, persists across reload
- [ ] Reviewer: optionally trigger the Hangfire job manually against a budget with an over-threshold category to see the bell notification (covered by unit tests here)

## Plan doc
`docs/plans/budget-alerts.md` — full design with phases, decisions, and risk mitigation.

## Screenshots
_Captured during verification — saved at repo root, drag into this description (GitHub image uploads are web-UI only):_
- `budget-alerts-settings-card.png` — the new Budget Alerts settings card
- `budget-alerts-settings-persisted.png` — threshold persisted after reload

## Summary by CodeRabbit

* **New Features**
  * Added budget-vs-actual alerts with in-app notifications when categories approach (default 80%) or are exceeded (deep-link included).
  * Added a “Budget Alerts” settings option to set your alert threshold (1–100%).
  * Alerts are deduplicated per budget period to prevent repeated pings.
* **Documentation**
  * Published a detailed specification and rollout plan for budget-vs-actual alerts.
* **Tests**
  * Added unit tests for daily processing, notification triggering, deduplication/idempotency, and key edge cases.

### CODE REVIEW

File: src/WebAPI/MyMascada.WebAPI/Program.cs [hunk 1]
```diff
@@ -453,6 +453,14 @@ static bool IsPersistentVolume(string markerPath)
...
     Hangfire.Cron.Daily(1, 0)); // Run daily at 1:00 AM
 
+// Runs at 1:30 AM — after the 1:00 AM expired-budget job has rolled periods
+// over, so alerts evaluate the current period, not the one that just ended.
+// Hangfire replaces the CancellationToken with its shutdown token at runtime.
+recurringJobManager.AddOrUpdate<MyMascada.Application.BackgroundJobs.IBudgetAlertJobService>(
+    "daily-budget-alert-check",
+    service => service.ProcessAllUsersAsync(default),
+    Hangfire.Cron.Daily(1, 30)); // Run daily at 1:30 AM
+
```
[the remark this answers is not in the document]
Comment (digaomatias) [reply src/WebAPI/MyMascada.WebAPI/Program.cs:468]: Skipping this one, with reasoning. The 1:30 AM offset isn't a correctness guarantee, agreed — but the failure mode here is low-severity and self-healing, and full job-chaining is disproportionate:

- The alert `groupKey` is scoped to the budget **period start**, not the run date. If the alert job ever evaluates a not-yet-rolled-over (expired) period, the worst case is one boundary-day alert keyed to the old period; once rollover runs, the new period has its own groupKey and alerts correctly on the next daily pass. No duplicate or permanently-wrong state.
- Rollover (`ExpiredBudgetJobService`) processes expired budgets in seconds-to-minutes for any realistic user base; 30 minutes is ample headroom.
- Gating one recurring job on another's completion (or merging the alert trigger into the rollover flow) adds real coupling/complexity for a once-per-day, self-correcting edge case — you flagged it as a 🏗️ heavy lift yourself.

If budget counts ever grow enough that rollover approaches the 30-min window, the right move is a continuation/`ContinueJobWith` chain — I'd rather do that when it's warranted than couple the jobs pre-emptively. Leaving as-is.

File: src/WebAPI/MyMascada.WebAPI/Program.cs [hunk 2]
```diff
@@ -453,6 +453,14 @@ static bool IsPersistentVolume(string markerPath)
     service => service.ProcessAllUsersAsync(null),
     Hangfire.Cron.Daily(1, 0)); // Run daily at 1:00 AM
 
+// Runs at 1:30 AM — after the 1:00 AM expired-budget job has rolled periods
+// over, so alerts evaluate the current period, not the one that just ended.
+// Hangfire replaces the CancellationToken with its shutdown token at runtime.
+recurringJobManager.AddOrUpdate<MyMascada.Application.BackgroundJobs.IBudgetAlertJobService>(
+    "daily-budget-alert-check",
+    service => service.ProcessAllUsersAsync(default),
+    Hangfire.Cron.Daily(1, 30)); // Run daily at 1:30 AM
```
[the remark this answers is not in the document]
Comment (digaomatias) [reply src/WebAPI/MyMascada.WebAPI/Program.cs:467]: Good catch — fixed in 5600e38, and more broadly than the schedule. The root cause was that `NotificationService.CreateNotificationAsync` *dropped* the notification entirely during quiet hours (no record, no groupKey), and since in-app is the only delivery channel implemented, it was lost permanently — so a fixed-time daily producer landing in a user's quiet window would never deliver.

Quiet hours now gate only real-time push delivery (documented at the future dispatch hook); the in-app record is always created so the user sees it when they next open the bell. This is the correct semantics for all notification types, not just budget alerts. The per-type inApp toggle and groupKey idempotency are unchanged. Added `NotificationServiceTests` covering created-during-quiet-hours plus regression coverage for the inApp toggle and dedup.
